packages/artifact/src/internal/requestUtils.ts

```diff
@@ -0,0 +1,74 @@
+import {IHttpClientResponse} from '@actions/http-client/interfaces'
+import {isRetryableStatusCode, isSuccessStatusCode, sleep} from './utils'
+import * as core from '@actions/core'
+
+export async function retry<T>(
+  name: string,
+  method: () => Promise<T>,
```

Comment:
Technically, this isn't necessarily a method:

```suggestion
  operation: () => Promise<T>,
```